src/main/scala/com/dwolla/rabbitmq/topology/LambdaHandler.scala

```diff
@@ -17,49 +13,38 @@ import fs2.INothing
 import io.circe.Encoder
 import natchez._
 import natchez.http4s.NatchezMiddleware
-import natchez.xray.XRay
 import org.http4s.client.{Client, middleware}
 import org.http4s.ember.client._
 import org.typelevel.log4cats.Logger
 import org.typelevel.log4cats.slf4j.Slf4jLogger
 
 import scala.annotation.nowarn
+import natchez.noop.NoopEntrypoint
 
 class LambdaHandler extends IOLambda[RabbitMQConfig, INothing] {
   val lambdaName = "RabbitMQ-Topology-Backup"
 
-  private implicit def kleisliLogger[F[_] : Logger, A]: Logger[Kleisli[F, A, *]] = Logger[F].mapK(Kleisli.liftK)
-
-  private implicit def kleisliLambdaEnv[F[_] : Functor, A, B](implicit env: LambdaEnv[F, A]): LambdaEnv[Kleisli[F, B, *], A] =
-    env.mapK(Kleisli.liftK)
-
   private def httpClient[F[_] : Async]: Resource[F, Client[F]] =
     EmberClientBuilder
       .default[F]
       .build
       .map(middleware.Logger[F](logHeaders = true, logBody = false))
 
-  private def resources[F[_] : Async] =
-    Resource.eval(Random.scalaUtilRandom[F]).flatMap { implicit random =>
-      (XRay.entryPoint[F](), KmsAlg.resource[F], httpClient[F]).tupled
-    }
+  override def handler = for {
+    ep <- Resource.pure(NoopEntrypoint[IO]())
+    kms <- KmsAlg.resource[IO]
+    http <- httpClient[IO]
+    logger <- Resource.eval(Slf4jLogger.fromName[IO](lambdaName))
+  } yield { implicit env =>
+    implicit val l = logger
```

Comment:
One day in Scala 3 we will make the logger implicit in the for-comprehension :)

Reply:
Actually this project already has better-monadic-for enabled, so we can do it here too; it's just not as pretty. 🙂 

```scala
implicit0(logger: Logger[F]) <- Resource.eval(Slf4jLogger.fromName[F](lambdaName))
```